includes/import/abstract-wc-product-importer.php

```diff
@@ -0,0 +1,941 @@
+
+				if ( 0 === $attachment_id && isset( $image['src'] ) ) {
+					$upload = wc_rest_upload_image_from_url( esc_url_raw( $image['src'] ) );
+
+					if ( is_wp_error( $upload ) ) {
+						if ( ! apply_filters( 'woocommerce_rest_suppress_image_upload_error', false, $upload, $product->get_id(), $images ) ) {
+							throw new Exception( $upload->get_error_message(), 400 );
+						} else {
+							continue;
+						}
+					}
+
+					$attachment_id = wc_rest_set_uploaded_image_as_attachment( $upload, $product->get_id() );
+				}
+
+				if ( ! wp_attachment_is_image( $attachment_id ) ) {
+					throw new Exception( sprintf( __( '#%s is an invalid image ID.', 'woocommerce' ), $attachment_id ), 400 );
+				}
+
+				if ( isset( $image['position'] ) && 0 === absint( $image['position'] ) ) {
+					$product->set_image_id( $attachment_id );
+				} else {
+					$gallery[] = $attachment_id;
+				}
+
+				// Set the image alt if present.
+				if ( ! empty( $image['alt'] ) ) {
+					update_post_meta( $attachment_id, '_wp_attachment_image_alt', wc_clean( $image['alt'] ) );
+				}
+
+				// Set the image name if present.
+				if ( ! empty( $image['name'] ) ) {
+					wp_update_post( array( 'ID' => $attachment_id, 'post_title' => $image['name'] ) );
+				}
+			}
+
+			if ( ! empty( $gallery ) ) {
+				$product->set_gallery_image_ids( $gallery );
+			}
+		} else {
+			$product->set_image_id( '' );
+			$product->set_gallery_image_ids( array() );
+		}
+
+		return $product;
+	}
+
+	/**
+	 * Save product shipping data.
+	 *
+	 * @param WC_Product $product Product instance.
+	 * @param array      $data    Shipping data.
+	 * @return WC_Product
+	 */
+	protected function save_product_shipping_data( $product, $data ) {
+		// Virtual.
+		if ( isset( $data['virtual'] ) && true === $data['virtual'] ) {
+			$product->set_weight( '' );
+			$product->set_height( '' );
+			$product->set_length( '' );
+			$product->set_width( '' );
+		} else {
+			if ( isset( $data['weight'] ) ) {
+				$product->set_weight( $data['weight'] );
+			}
+
+			// Height.
+			if ( isset( $data['height'] ) ) {
+				$product->set_height( $data['height'] );
+			}
+
+			// Width.
+			if ( isset( $data['width'] ) ) {
+				$product->set_width( $data['width'] );
+			}
+
+			// Length.
+			if ( isset( $data['length'] ) ) {
+				$product->set_length( $data['length'] );
+			}
+		}
+
+		// Shipping class.
+		if ( isset( $data['shipping_class_id'] ) ) {
+			$shipping_class_term = get_term_by( 'id', wc_clean( $data['shipping_class_id'] ), 'product_shipping_class' );
+
+			if ( $shipping_class_term ) {
+				$product->set_shipping_class_id( $shipping_class_term->term_id );
+			}
+		}
+
+		return $product;
+	}
+
+	/**
+	 * Save downloadable files.
+	 *
+	 * @param WC_Product $product    Product instance.
+	 * @param array      $downloads  Downloads data.
+	 * @param int        $deprecated Deprecated since 3.0.
+	 * @return WC_Product
+	 */
+	protected function save_downloadable_files( $product, $downloads ) {
```

Comment:
Should be handled during import. See my comment above about attributes. We will have to make some changes to the parser, though, to map downloads into a format that this can parse. I've detailed them below in a comment.